### ISSUE TITLE
Multilingual Support for paraphrase-MiniLM-L3-v2 and paraphrase-MiniLM-L6-v2

### ISSUE THREAD

**Issue by duraisanthiya** [UKPLab/sentence-transformers#3300]
I wanted to check if there are any plans to add multilingual support for the paraphrase-MiniLM-L3-v2 and paraphrase-MiniLM-L6-v2 models. If this feature is not planned, could you provide guidance on how I might extend these models to support multiple languages?

**Comment by duraisanthiya** [comment]
Hi @nreimers and @IrynaGurevych , any leads on this or insights into the pretraining of paraphrase-multilingual-MiniLM-L12-v2 would be highly helpful.

**Comment by tomaarsen** [comment]
Hello!

Apologies for the delay.
I don't have plans currently to train those models. However, I can share some details about how they could be trained: [Sentence Transformers > Training Examples > Model Distillation](https://sbert.net/examples/sentence_transformer/training/distillation/README.html). Notably the [model_distillation_layer_reduction.py](https://github.com/UKPLab/sentence-transformers/tree/master/examples/sentence_transformer/training/distillation/model_distillation_layer_reduction.py) script should come in handy: it can be used to shrink the layers of a model while keeping as much of the original performance as possible.

Some requires changes that would need to be made:
1. Update the `teacher_model_name` to `"paraphrase-multilingual-MiniLM-L12-v2"`
2. When updating the layers, ensure that `auto_model.encoder.layer` is the correct way to get to the `ModuleList` with the model layers. Depending on the architecture, it may be called something else.
3. The training (and evaluation) dataset must be updated to something multilingual for the best performance. If you want to use this model for retrieval, then perhaps you can use some samples from https://huggingface.co/datasets/nthakur/swim-ir-monolingual, and if you want it for e.g. paraphrase detection, then perhaps you can use texts like these: https://huggingface.co/datasets/sentence-transformers/parallel-sentences-talks.

In short, this script uses a distillation loss: [MSELoss](https://sbert.net/docs/package_reference/sentence_transformer/losses.html#mseloss) to be specific. You can then pass a training dataset with a text column and a `label` column that has the embedding from the teacher model (paraphrase-multilingual-MiniLM-L12-v2). The student model (i.e. just the teacher model but with less layers) will then be trained to try and match the teacher embeddings.

- Tom Aarsen

**Comment by esafak** [comment]
Perhaps you could develop the script and leave training to the community then?

**Comment by duraisanthiya** [comment]
Thanks for sharing this, @tomaarsen! I’ll definitely explore this and see how it fits for multilingual support. Appreciate the detailed guidance!
